src/View/Widget/DateTimeNativeWidget.php

```diff
@@ -0,0 +1,130 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         3.0.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\View\Widget;
+
+use Cake\View\Form\ContextInterface;
+use Cake\View\StringTemplate;
+use DateTime;
+use Exception;
+use RuntimeException;
+
+/**
+ * Input widget class for generating a date time input widget.
+ *
+ * This class is intended as an internal implementation detail
+ * of Cake\View\Helper\FormHelper and is not intended for direct use.
+ */
+class DateTimeNativeWidget implements WidgetInterface
```

Comment:
I wonder if we should just make native date inputs the default for 4.0. HTML5 date input support has gotten much better with evergreen browsers and not having to support multiple selectboxes would let us save some complexity internally as well.

Reply:
I am fine with making native datetime inputs default in 4.0 and dropping the selects altogether. In practice most sites/apps use JS based pickers anyway for better consistency and usability.

Reply:
We should retain the array marshalling for datetimes for backwards compatibility.  I bet somewhere someone is using array dates in an API.

Reply:
Keeping datetime array marshaling sounds reasonable.

So should I reopen this against `4.x` instead, with the helper and widget code for selects removed?

Reply:
I think that is a good idea. We could move the old widgets into a plugin that folks can use if they really need that behavior.

Reply:
Okay, closing this then. Will open a new PR targeting 4.x soon.

Reply:
What would you have liked to see?

Reply:
I prefer using dropdown to select the year for birthday, which can allow to select "9999" for the year if someone prefers say his year of birth. Therefore for birthday I like the select dropdown version.

Reply:
I think `year`, and `month` inputs could stay as select boxes. The native HTML inputs are better for `date` and `time` values though.

Reply:
If we are going to switch to native inputs then by default they should be used for all available types. There's no HTML input for `year` so keeping the select box for `year` is fine but there is one for `month` so for that native input should be used.